### DISCUSSION TITLE
Not able to process message after upgrade from 3.8.1 to 4.4.1

### DISCUSSION THREAD

**Discussion by Jaysarun** [nextgenhealthcare/connect#6069] [Q&A] [answered]
Recently we have upgraded our Mirth connect from 3.8.1 to 4.4.1. After the upgrade we are not able to process through the channels and it is throwing the error "throw "Unknown or unsupported version of Mirth Connect." before importing the channel into the new version we have edited to new version. Any idea or suggestion to resolve this?

![image](https://github.com/nextgenhealthcare/connect/assets/155644178/be69fc7c-8af6-49ba-b5a0-fbf64950a80a)

**Comment by pacmano1** [comment]
Did you already post this?

**Reply by Jaysarun** [reply]
yes, but I didn't get any resolution for this and stuck for a while now

**Reply by pacmano1** [reply]
You did not format your code as requested on the other post.

The code you did post does not contain the string shown in your screen shot, so look for that string in code templates, deploy scripts and global deploy scripts.

**Reply by Jaysarun** [reply]
Okay, I will try it. Thank u!

**Comment by rjkroll** [comment]
Looks like you are having the same issue you did 3 weeks ago?   Have you tried anything suggested in this tread?  

https://github.com/nextgenhealthcare/connect/discussions/6049

Like previously mentioned, the error looks to be coming from custom code you inserted.   Keep stripped out the code, etc. until you find the source of the issue.

**Reply by Jaysarun** [reply]
Thanks for response, I do tried without my custom codes and it is working fine but my custom codes (JS) are essential to transform the data before inserting into the tables. In the older version channels are running fine with JS codes I have.

**Comment by jonbartels** [comment]
![image](https://github.com/nextgenhealthcare/connect/assets/7669292/e936a253-ee43-429d-a4da-58fe08a9caea)

The problem is the code that you or another engineer at your company wrote in a Mirth Connect filter transformer. You need to look at the ENTIRE block of code and not just the error message Mirth shows when it tries to process a message.

**Reply by Jaysarun** [reply]
Noted Jon, I will go through it again and let you know. Thanks!

**Reply by Jaysarun** [reply]
Hi Guys,
I am able to fix the above issue I have mentioned earlier. Thanks for your inputs.

**Reply by pacmano1** [reply]
@Jaysarun what was the fix?  You found the offending code???

If you are going to ask questions here and expect answers, share your answers also.

**Reply by Jaysarun** [reply, marked as answer]
@pacmano1 There was one specific code designed for the Mirth Connect v2 & v3 versions.  I took out that block but afterwards I am facing another issue related to Parsing. I will post it separately.

Thanks,
Arunprasanna Jay S
